compel/arch/x86/src/lib/infect.c

```diff
@@ -306,6 +306,17 @@ static int corrupt_extregs(pid_t pid)
 	} else {
 		struct iovec iov;
 
+		/*
+		 * While using PTRACE_SETREGSET the kernel checks that
+		 * "Reserved bits in MXCSR must be zero."
+		 * if (mxcsr[0] & ~mxcsr_feature_mask)
+		 *	return -EINVAL;
+		 *
+		 * As the mxcsr_feature_mask depends on the CPU the easiest solution for
+		 * this error injection test is to set mxcsr just to zero.
+		 */
+		ext_regs.i387.mxcsr = 0;
+
 		validate_random_xstate((void *)&ext_regs);
```

Comment:
Could you move it into `validate_random_xstate()`? AFAIR, it's the helper to force random xstate to be valid.

Reply:
I see you just updated your comment. I was thinking similar. As this is only for error injection tests we could ignore the mask.

Reply:
> Sure. Do you see a problem with my change that I am ignoring the mxcsr_feature_mask and setting the complete mxcsr to zero?

I don't. I think it makes this force-injection test a bit less precise - we won't check that mxcsr flags that may have been corrupted by the parasite dumping are restored on the cure of the task. But we still corrupt everything else on xstate, so I don't see how criu could revert any possible corruption, without reverting it also for mxcsr register.
So, I think the test is still fine.

> I see you just updated your comment. I was thinking similar. As this is only for error injection tests we could ignore the mask.

Yeah, pressed "return", forgot to finish part of the comment :)

Reply:
Moved everything to `validate_random_xstate()`.